src/components/Capture/withCameraDetection.js

```diff
@@ -2,10 +2,17 @@ import { h } from 'preact'
 import { PureComponent } from 'preact-compat'
 import { checkIfHasWebcam } from '../utils'
 
+// It's unlikely that whether or not our device has a camera will change, so we
+// speed ourselves up slightly by assuming that the result was whatever our
+// previous value was (but we still actually check, and update the value correctly)
+// If we are incorrect, then the user will simply see a flicker of a fallback,
+// or an empty camera stream, before the correct screen once more
+let previousHasCameraCheck = null
```

Comment:
What happened if the webcam is disconnected mid-flow?

Reply:
We still _actually_ check the webcam support, throughout. We only use this variable as a means of getting a more sane "start" value (i.e. we assume you have a webcam since you did before, but still _actually_ check that you do, and then correct ourselves if we were wrong)

But I'm assuming this idea of a user disconnecting mid-flow is actually fairly rare, in real use-cases (hence we should still allow it to happen, but not optimise for it)

Reply:
Although messing about a bit more, it looks like the way that the forwarding to the cross-device flow when no camera support is enabled works, then our caching is fine when `true` --> `false`
But when `false` --> `true`, you're actually stuck with the code always assuming you don't have a webcam. There are ways we can work around this, but it'll end up just causing more hassle than it's worth - so will just remove the idea of caching here at all